src/internal/m365/collection/drive/collections.go

```diff
@@ -47,9 +47,16 @@ const restrictedDirectory = "Site Pages"
 type Collections struct {
 	handler BackupHandler
 
-	tenantID      string
+	tenantID string
+
+	// resourceOwner is the owner of the top level resource
 	resourceOwner string
 
+	// serviceOwner is the owner of the service we are backing up.
+	// serviceOwner and resourceOwner will be only when backing up a
+	// SharePoint site associated with a Group.
+	serviceOwner string
```

Comment:
I'd like to challenge this implementation a bit.  The resource and service owners are so often the same that I worry thier usage will be perpetually unclear.  That confusion could lead to accidental and difficult-to-pinpoint bugs.  For example, I think the choice here to store the metadata under a path that matches sharepoint formatting instead of matching the standard groups formatting both feels accidental and could lead to unexpected side effects.

A more holistic implementation would be to extend the handler with a func like `metadataDir() path.Elements`, and modify `graph.NewMetadataEntry` to accept an Elements slice or Path instead of only a string.  This way we:
* allow any backup producer to arbitrarily nest metadata locations (reusable)
* maintain a single owner within the collection (simple)
* maintain the same prefix pathing for all data in the collection (consistent)

Reply:
Yup, that makes sense and was the plan long term. To be frank this was more of a short term fix, but I guess it does not make much sense to go this route. I think I will just implement passing partial paths downstream as I am starting with it anyways.

Reply:
> I think I will just implement passing partial paths downstream as I am starting with it anyways.

Can you clarify that for me?  Does that mean you're going to close this PR and shift the work elsewhere?  Amend this PR in a follow-up PR?  Amend this PR directly?

Reply:
Closing in favor of https://github.com/alcionai/corso/pull/4156